src/wallet/home_cards/manager.rs

```diff
@@ -0,0 +1,378 @@
+                    })
+                    .await;
+                Ok(())
+            }
+            Err(CommonError::FailedAccessingHomeCards) => {
+                let default_cards = HomeCards::from_iter([HomeCard::Connector]);
+                let updated_cards = self
+                    .update_cards(|write_guard| {
+                        self.insert_cards(write_guard, default_cards)
+                    })
+                    .await?;
+                self.save_cards(updated_cards).await
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Handles a deferred deep link by parsing it, saving the generated home cards to storage and notifying the observer.
+    /// Notifies `HomeCardsObserver`.
+    #[uniffi::method]
+    pub async fn handle_deferred_deep_link(
+        &self,
+        encoded_value: String,
+    ) -> Result<()> {
+        let deep_link_cards = self.parser.parse(encoded_value).await?;
+        let updated_cards = self
+            .update_cards(|write_guard| {
+                self.insert_cards(write_guard, deep_link_cards)
+            })
+            .await?;
+        self.save_cards(updated_cards).await
+    }
+
+    /// Removes a specified home card, save the updated home cards to storage and notifies the observer.
+    /// Notifies `HomeCardsObserver`.
+    #[uniffi::method]
+    pub async fn remove_card(&self, card: HomeCard) -> Result<()> {
+        let updated_cards = self
+            .update_cards(|write_guard| {
+                write_guard.remove_id(&card);
+            })
+            .await?;
+        self.save_cards(updated_cards).await
+    }
+}
+
+impl HomeCardsManager {
+    /// Updates home cards both in-memory and in storage after applying `f` function, then notifies the observer.
+    async fn update_cards<F>(&self, f: F) -> Result<HomeCards>
+    where
+        F: FnOnce(&mut RwLockWriteGuard<HomeCards>),
+    {
+        let mut write_guard = self
+            .cards
+            .write()
+            .map_err(|_| CommonError::FailedUpdatingHomeCards)?;
+
+        f(&mut write_guard);
+
+        let updated_cards = write_guard.clone();
+        updated_cards.sort();
+
+        self.observer.handle_cards_update(updated_cards.clone());
+        Ok(updated_cards)
+    }
+
+    fn insert_cards(
+        &self,
+        write_guard: &mut RwLockWriteGuard<HomeCards>,
+        cards: HomeCards,
+    ) {
+        cards.into_iter().for_each(|card| {
+            if write_guard.try_insert_unique(card).is_ok() {
+                debug!("Home card inserted");
+            } else {
+                debug!("Home card insert failed");
+            }
+        })
+    }
+}
+
+impl HomeCardsManager {
+    /// Loads the home cards from storage.
+    async fn load_cards(&self) -> Result<HomeCards> {
+        let cards_bytes = self
+            .cards_storage
+            .load_cards()
+            .await?
+            .ok_or(CommonError::FailedAccessingHomeCards)?;
```

Comment:
This error is more like there are no cards in the storage, no? Is it an error if there are no cards?

Reply:
Most likely yes, but it's for any error returned by the `load_cards` trait function. Should I rename it to `HomeCardsNotFound` to match the equivalent error mapped from `session_storage.load_session`? Please let me know if you think of a better name for this kind of error.

Reply:
`HomeCardsNotFound` sounds good.